### PR TITLE
338 review general activity

### PR DESCRIPTION
**Pull request by wbaccinelli**
Fixes: https://github.com/PerfectFit-project/virtual-coach-issues/issues/312

The activities timing has been defined in the `resources_timing.json`. Here, for each activity the following template has been used:
```
      "resource_id": int,
      "category": str,
      "always_available":bool,
      "phases":[
         {
            "phase":str,
            "always_available":bool,
            "time_unit":str,
            "available":[
               int
            ],
            "mandatory":[
               int
            ]
         },
      ]
```

the `resource_id` is the id of the specific resource, as reported in the DB, and defined in the `activities.csv` file

`category` refers to the type of activity, which can be one of the following:
- 'self-related'
- 'educational'
- 'practical',
- 'reinforcement'
- 'motivational'

`always_available` is set to true if the resource can be used in every moment during the intervention/the specific phase. 

`available` is a list of days/weeks during which the resource can be used.

`mandatory` is a list of days/weeks during which the resource must be used.

`time_unit` indicates whether the numbers in `available` or in `mandatory` are days or weeks.

The json file can be used in other packages, using the `get_timing` function implemented in the `helper_functions`

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
         'virtual_coach_db.dbschema': './dbschema'
     },
     include_package_data=True,
+    package_data={'': ['../resources/*.json']},
```
Comment (raar1) [inline setup.py:16]: You intend to have more resources json files in future?

File: helper/helper_functions.py [hunk 2]
```diff
@@ -38,3 +40,13 @@ def get_db_session(db_url=DB_URL_DEFAULT):
     session = session_maker()
 
     return session
+
+
+def get_timing():
+    string_json = importlib_resources.files('virtual_coach_db').joinpath(
+        'resources/resources_timing.json').read_text()
+
+    timing = json.loads(string_json)
```
Comment (raar1) [inline helper/helper_functions.py:49]: The one thing I would expect to see here is some sort of validation of the json file to make sure it has the right structure, or that values are within accepted limits (or that phases correspond to allowed values etc). I'm not sure how thoroughly you would want to validate it but at least some might help catch bugs later

Comment (raar1) [reply helper/helper_functions.py:49]: For example, the `jsonschema` python library looks relatively easy to make and apply a schema to validate the json structure. For example, you can write a schema that looks like e.g.
```
{
  "type": "object",
  "properties": {
    "first_name": { "type": "string" },
    "last_name": { "type": "string" },
    "birthday": { "type": "string", "format": "date" },
    "address": {
      "type": "object",
      "properties": {
        "street_address": { "type": "string" },
        "city": { "type": "string" },
        "state": { "type": "string" },
        "country": { "type" : "string" }
      }
    }
  }
}
```
(an example from their docs)

and then apply that to the resources json

**Review by raar1 — APPROVED** [review]
Good work! :sunglasses:  I've left a comment below

File: helper/helper_functions.py (hunk 2 above)
Comment (raar1) [reply helper/helper_functions.py:49]: Haven't used it myself though so... beware... :stuck_out_tongue:

File: setup.py (hunk 1 above)
Comment (wbaccinelli) [reply setup.py:16]: No, I'm restricting this to just the file we are using now.

File: helper/helper_functions.py (hunk 2 above)
Comment (wbaccinelli) [reply helper/helper_functions.py:49]: Oh yes, this is a great idea!

**Comment by wbaccinelli** [comment]
@raar1 I have added a json schema verification with a test action. Can you quickly check that what I did as a workflow makes any sense at all? :)

**Pull request merged**